src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface State {
+  key: string | null,
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    key: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyPress);
+  }
+
+  handleKeyPress = (event: KeyboardEvent) => {
+    this.setState({ key: event.key });
+  };
+
+  componentDidUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyPress);
+  }
```

Comment:
Try to organize similar things
```suggestion
componentDidUnmount(): void {
    document.removeEventListener('keyup', this.handleKeyPress);
  }

  handleKeyPress = (event: KeyboardEvent) => {
    this.setState({ key: event.key });
  };
```

Reply:
Works with componentWillUnmount though